geniza/corpus/management/commands/tei_to_annotation.py

```diff
@@ -0,0 +1,347 @@
+                )
+            # if we need more canvases than we have available,
+            # generate local canvas ids
+            if num_canvases > len(iiif_canvases):
+                # document manifest url is /documents/pgpid/iiif/manifest/
+                # create canvas uris parallel to that
+                canvas_base_uri = doc.manifest_uri.replace("manifest", "canvas")
+                for i in range(num_canvases - len(iiif_canvases)):
+                    canvas_uri = "%s%d/" % (canvas_base_uri, i + 1)
+                    iiif_canvases.append(canvas_uri)
+
+            # NOTE: pgpid 1390 folio example; each chunk should be half of the canvas
+            # (probably should be handled manually)
+            # if len(html_chunks) > len(iiif_canvases):
+            #     self.stdout.write(
+            #         "%s has more html chunks than canvases; skipping" % xmlfile
+            #     )
+            #     continue
+
+            # start attaching to first canvas; increment based on chunk label
+            canvas_index = 0
+
+            # remove all existing annotations associated with this
+            # document and source so we can reimport as needed
+            existing_annos = annotation_client.search(
+                source=footnote.source.uri, manifest=doc.manifest_uri
+            )
+            # FIXME: this is probably problematic for transcriptions currently
+            # split across two TEi files...
+            # maybe filter by date created also, so we only remove
+            # annotations created before the current script run?
+            if existing_annos:
+                print(
+                    "Removing %s existing annotation(s) for %s on %s "
+                    % (len(existing_annos), footnote.source, doc.manifest_uri)
+                )
+                for anno in existing_annos:
+                    annotation_client.delete(anno["id"])
+
+            for i, block in enumerate(html_blocks):
+                # if this is not the first block and the label suggests new image,
+                # increment canvas index
+                if i != 0 and tei.label_indicates_new_page(block["label"]):
+                    canvas_index += 1
+
+                # get the canvas uri for this section of text
+                annotation_target = iiif_canvases[canvas_index]
+
+                anno = new_transcription_annotation()
+                # link to digital edition footnote via source URI
+                anno["dc:source"] = footnote.source.uri
+
+                anno.target.source.id = annotation_target
+                anno.target.source.partOf.type = "Manifest"
+                anno.target.source.partOf.id = doc.manifest_uri
+
+                # apply to the full canvas using % notation
+                # (using nearly full canvas to make it easier to edit zones)
+                anno.target.selector.value = "xywh=percent:1,1,98,98"
```

Comment:
I was running into trouble using this selector on annotorious, were you able to test it?

Reply:
I had to do some experimenting to get a version that annotorious rendered properly; this is the one that worked.